src/RewardsDistributor.sol

```diff
@@ -70,7 +71,12 @@ contract RewardsDistributor is IRewardDistributor {
                 "Collateral does not match the rewards token"
             );
         }
-        payoutToken.safeTransfer(payoutTarget_, payoutAmount_);
+
+        // payoutAmount_ is always in 18 decimals precision, adjust actual payout amount to match payout token decimals
+        uint256 systemPrecision = 10 ** 18;
+        uint256 distributorPrecision = 10 ** IERC20(payoutToken).decimals();
```

Comment:
to save gas, since payout token doesn't change in an ERC20, the payout token decimals should be saved within this contract during construction (also nice becuase it serves as a sanity check for payout token)

Reply:
✅ 
Resolved `precision` in constructor and saved `SYSTEM_PRECISION` as const